### PR TITLE
[ONL-6724] - migrate button dropdown component

### PR DESCRIPTION
**Pull request by DisegnAle**
migrated component to vue3

### CODE REVIEW

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue [hunk 1]
```diff
@@ -5,31 +5,31 @@
     :data-test="$attrs['data-test'] ? `${$attrs['data-test']} ec-btn-dropdown` : 'ec-btn-dropdown'"
   >
     <ec-btn
-      :is-disabled="isDisabled"
+      :is-disabled="props.isDisabled"
```
Comment (mcibique) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.vue:8]: Props are automatically available in the template, so you don't have to update their references in the template. You can remove all occurrences of `props.*` from the template.

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue [hunk 2]
```diff
@@ -44,55 +44,57 @@
...
-  data() {
+  listDataTest: {
+    type: String,
+  },
+});
+
+const emit = defineEmits(['click', 'change']);
+
+const isOpen = ref(false);
+const popoverOptions = ref({
```
Comment (mcibique) [inline src/components/ec-btn-dropdown/ec-btn-dropdown.vue:74]: I noticed that this object is static and can't change in the future. We can optimize that either by removing the `ref()` so the reactivity doesn't have to keep track of it, or we can just inline the object into the template `:popover-options="{ autoSize: 'min', placement: 'bottom-end', }"`

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue (hunk 1 above)
Comment (DisegnAle) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.vue:8]: corrected

File: src/components/ec-btn-dropdown/ec-btn-dropdown.vue (hunk 2 above)
Comment (DisegnAle) [reply src/components/ec-btn-dropdown/ec-btn-dropdown.vue]: corrected

**Review by mcibique — APPROVED** [review]

**Pull request merged**